formula-coroutines/src/main/java/com/instacart/formula/coroutines/FlowAction.kt

```diff
@@ -70,12 +65,9 @@ interface FlowAction<Event> : Action<Event> {
 
     fun flow(): Flow<Event>
 
-    val scope: CoroutineScope
-
     override fun start(send: (Event) -> Unit): Cancelable? {
-        val job = flow()
-            .onEach { send(it) }
-            .launchIn(scope)
-        return Cancelable(job::cancel)
+        val scope = MainScope()
+        flow().onEach { send(it) }.launchIn(scope)
+        return Cancelable(scope::cancel)
```

Comment:
I just remembered one thing: when flow terminates (completes), does it cause scope cancellation?

add on [onCompletion](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/on-completion.html)?